### PR TITLE
Test/cicd code quality and formatting

### PR DESCRIPTION
**Pull request by 444B**
# Pull Request Template

## TL;DR
Provide a brief summary of the changes made.
- added code quality checks
- used black, isort, flake8, mypy and bandit to clean up the code, resolving 3 security issues
- bumped minimum python version to 3.9 to resolve minimum import issues in numpy 1.26.4
- issues in code checking will continue on a failure when in a branch that begins with /test but will fail when being published to main and if the issues still exist
- lots of code cleanup and removal of whitespace
- introduction of tests/run_checks.sh in order to replicate code checks locally what would normally take place in a Github Action.
- slowly starting to write unit tests with pytest

## Problem Solved
1. Describe the problem your changes address. 
2. Please mention the issue number if relevant
- code was not following a standard
- this led to security issues in the code, such as making use of subprocces() where not necessary

## Manual Testing
- [X] You have manually tested all changes to ensure they work as intended. Replace the [ ] with a [X] if True.

**How was this tested?**
Describe the tests that you ran to verify your changes.
- run locally

## Additional Notes
Include any additional information that might be relevant for reviewers.
-

### CODE REVIEW

**Comment by 444B** [comment]
will merge with the failed build-upload-push since that was a failure in test env. I think
